.circleci/scripts/git-diff-develop.ts

```diff
@@ -0,0 +1,94 @@
+import { promisify } from 'util';
+import { exec as execCallback } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+const exec = promisify(execCallback);
+
+/**
+ * Fetches the git repository with a specified depth.
+ *
+ * @param {number} depth - The depth to use for the fetch command.
+ * @returns {Promise<boolean>} True if the fetch is successful, otherwise false.
+ */
+async function fetchWithDepth(depth: number): Promise<boolean> {
+  try {
+    await exec(`git fetch --depth ${depth} origin develop`);
+    await exec(`git fetch --depth ${depth} origin ${process.env.CIRCLE_BRANCH}`);
+    return true;
+  } catch (error: any) {
+    console.error(`Failed to fetch with depth ${depth}:`, error.message);
+    return false;
+  }
+}
+
+/**
+ * Attempts to fetch the necessary commits until the merge base is found.
+ * It tries different fetch depths and performs a full fetch if needed.
+ *
+ * @throws {Error} If an unexpected error occurs during the execution of git commands.
+ */
+async function fetchUntilMergeBaseFound() {
+  const depths = [1, 10, 100];
+  for (const depth of depths) {
+    console.log(`Attempting git diff with depth ${depth}...`);
+    await fetchWithDepth(depth);
+
+    try {
+      await exec(`git merge-base origin/HEAD HEAD`);
+      return;
+    } catch (error: any) {
+      if (error.code === 1) {
```

Comment:
Nit: This is what I would use to replace `any` here:
```suggestion
    } catch (error: unknown) {
      if (
        error instanceof Error &&
        hasProperty(error, 'code') &&
        error.code === 1
      ) {
```